lib/facter/collectd_discoverd_osds.rb

```diff
@@ -0,0 +1,18 @@
+# Fact: collectd_discoverd_osds
+#
+# Purpose: discover ceph osds
+#
+# Resolution:
+#
+# Caveats:  not well tested
+#
+require 'puppet'
+
+if File.directory?('/var/lib/ceph/osd')
```

Comment:
The `File.directory?()` call is because Ruby uses forward slashes for path separators on any OS but I don't think that that's the path for Windows for example, if someone managed to pull that off.